hpx/parallel/algorithms/find.hpp

```diff
@@ -34,8 +34,8 @@ namespace hpx { namespace parallel { inline namespace v1
     namespace detail
     {
         /// \cond NOINTERNAL
-        template <typename FwdIter>
-        struct find : public detail::algorithm<find<FwdIter>, FwdIter>
+        template <typename Iter>
+        struct find : public detail::algorithm<find<Iter>, Iter>
```

Comment:
I had changed the name because the template was clashing with the FwdIter declaration in parallel. I can keep this as is or change this to FwdIter and the one in parallel to FwdIter1.